### ISSUE TITLE
nuclide-diagnostics-store Doesn't handle Linter v1 providers with no name

### ISSUE THREAD

**Issue by Arcanemagus** [facebook-atom/atom-ide-ui#12]
It appears that `nuclide-diagnostics-store` can't handle Linter v1 providers without a name specified. Although an unfortunate oversight in that older API, it should still be supported here as there are providers out there that are still on this older version of the API.

As a note, the error message also seems to be missing a space in the `linterundefined` part:
![image](https://user-images.githubusercontent.com/427137/30349798-1b7edab4-97c9-11e7-8cd1-71b11558c52a.png)
```
nuclide-diagnostics-store found problems with a linterundefined. Diagnostic messages from that linter will be unavailable.
  * provider must have a name
```

**Comment by hansonw** [comment]
Interesting... OK, we'll fix this and give them a default name

**Comment by Arcanemagus** [comment]
I'm also trying to figure out which provider it was so I can get it fixed there 😉.

**Comment by Arcanemagus** [comment]
Turns out it was `linter-rails-best-practices` in this case: https://github.com/masolin/linter-rails-best-practices/pull/4.

**Comment by hansonw** [comment]
Fixed by 2eb54f911e7971dc6308aa6d31b9d5ed856ef2fb
